File: pset6/sentimental-mario-less/mario.py
# Initializing get_height function
def get_height():
    # Starting an infinte loop that is manually broken
    while True:
        # Testing for errors
        try:
            height = int(input("Height: "))
            # If check passes, no error occurred, break loop
            if height > 0 and height < 9:
                break
            # Else if error occurs, print this
        except ValueError:
            print("That's not an integer!")
    # Returning value
    return height

# assuming an input of 4 for testing
height = get_height()

    # i = 0/1/2/3
for i in range(0, height):
    # emptySpace = 3/2/1/0
    emptySpace = height - 1 - i
    # hashtags = 1/2/3/4
    hashtags = i + 1
    print(" " * emptySpace + "#" * hashtags)

# Or

for i in range(height):
    # First loop to make up the rows of the pyramid
    for j in range(height - i - 1):
        # Second loop to make up the number of empty spaces
        print(" ", end="")
    # Printing empty space to allign the pyramid to the right
    for k in range(i + 1):
        # Third loop to make up the number of hashes/"#"
        print("#", end="")
    print()
    # Adding new line

File: pset1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = get_int("What height should the pyramid be?\n");
    while (height <= 0 || height > 8)
    // height should be a positive number that's not higher than 8
    {
        height = get_int("What height should the pyramid be?\n");
    // looping to repeat the question if input is incorrect
    }
    for (int i = 0; i < height; i++)
    // first loop to make up the rows of the pyramid
    {
        for (int b = 1; b < height - i; b++)
    // second loop to make up the number of empty spaces
        {
            printf(" ");
    // printing empty space to allign the pyramid to the right
        }
        for (int a = 0; a <= i; a++)
    // third loop to make up the number of hashes/"#"
        {
            printf("#");
    //printing a hash on the same row for every loop in the cycle
        }
        printf("\n");
    // printing new row after the hash loop to separate the pyramid tiers
    }
}